rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -50,6 +52,46 @@ type Resolver = resolver::Resolver;
 pub enum TunnelManagerError {
     #[fail(display = "Port Error: {:?}", _0)]
     PortError(String),
+    #[fail(display = "Invalid state")]
+    InvalidStateError,
+}
+
+/// Action that progresses the state machine
```

Comment:
what does this provide that just setting the enum doesn't?

Reply:
My idea was to differentiate actions that could be applied to a tunnel, from a state itself. More like an "event" that happens.

Reply:
The question here is if that will simplify anything.

Reply:
I could see it being useful in very complicated state machines. Not sure we're at that level